### PR TITLE
Full match in client and server for Module Standard

### PR DESCRIPTION
**Pull request by idanstark42**
This PR should make this package as a wrap for `axios`, client and server, matching the Module Standard.
* Adding correlation only on server side, so it will work in client side.
* Adding client-id
* Adding logging only on server side

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -46,5 +46,10 @@
   "peerDependencies": {
     "@first-lego-league/ms-correlation": "^0.0.5",
     "axios": "^0.19.0"
+  },
+  "dependencies": {
+    "@first-lego-league/ms-logger": "^2.0.1",
```
Comment (2roy999) [inline package.json]: optional dependency

File: package.json [hunk 2]
```diff
@@ -46,5 +46,10 @@
   "peerDependencies": {
     "@first-lego-league/ms-correlation": "^0.0.5",
```
Comment (2roy999) [inline package.json]: optinal dependency

File: index.js [hunk 3]
```diff
@@ -1,17 +1,27 @@
 const axios = require('axios')
+const isServer = require('detect-node')
```
Comment (2roy999) [inline index.js]: I don't really like the design of this. A alternative design would be:
1. make the same file with a `createClientCreator`  function (funny name but that what it is). This function would get a `correlate` and logging functions (which might do nothing). 
2. make two main files one for browser and one for server. In the server we will pass the correlate function and the logging functions and in the browser file we will pass empty function (for now, in the future we might change those function).
3. In the package json you can add the `browser` field to change the index file when running in the browser.

Comment (idanstark42) [reply index.js]: Why not use the `is-server` package? I think if we can encapsulate it from the package's user, we should.
About the design. this package is a factory, and your'e describing a builder. I don't really mind, but in this design I can make any client in a single line, and in the builder design you'll sometimes need several. This is the main reason I prefer this one.

Comment (2roy999) [reply index.js]: I'm not talking about builder, I'm talking about a factory of the factory. The idea is to create polymorphism for the `correlate` and the logging functions.

Comment (idanstark42) [reply index.js]: They have, they get options as an argument.
If you create an object, of which the sole purpose is to create another object, it's a builder. But the definitions didn't matter. I still need a few lines to create a correlated independent client

**Review by 2roy999 — APPROVED** [review]

**Pull request merged**